packages/hardware-trezor/test/TrezorKeyAgent.test.ts

```diff
@@ -0,0 +1,234 @@
+    poolParameters: {
+      cost: '1000',
+      margin: {
+        denominator: '5',
+        numerator: '1'
+      },
+      metadata: {
+        hash: '0f3abbc8fc19c2e61bab6059bf8a466e6e754833a08a62a6c56fe0e78f19d9d5',
+        url: 'https://example.com'
+      },
+      owners: [
+        {
+          stakingKeyPath: [
+            util.harden(CardanoKeyConst.PURPOSE),
+            util.harden(CardanoKeyConst.COIN_TYPE),
+            util.harden(0),
+            2,
+            0
+          ]
+        }
+      ],
+      pledge: '10000',
+      poolId: 'cb0ec2692497b458e46812c8a5bfa2931d1a2d965a99893828ec810f',
+      relays: [
+        {
+          ipv4Address: '127.0.0.1',
+          port: 6000,
+          type: 0
+        },
+        {
+          hostName: 'example.com',
+          port: 5000,
+          type: 1
+        },
+        {
+          hostName: 'example.com',
+          type: 2
+        }
+      ],
+      rewardAccount: 'stake1u89sasnfyjtmgk8ydqfv3fdl52f36x3djedfnzfc9rkgzrcss5vgr',
+      vrfKeyHash: '8dd154228946bd12967c12bedb1cb6038b78f8b84a1760b1a788fa72a4af3db0'
+    },
+    type: Trezor.PROTO.CardanoCertificateType.STAKE_POOL_REGISTRATION
+  };
+
+  const requiredSigner = {
+    keyPath: [util.harden(CardanoKeyConst.PURPOSE), util.harden(CardanoKeyConst.COIN_TYPE), util.harden(0), 2, 0]
+  };
+
+  const validPlutusTx = {
+    ...simpleTx,
+    collateralInputs: [txInWithPath]
+  };
+  // END of Transformed / mapped data
+
+  describe('getSigningMode', () => {
+    it('can detect ordinary transaction signing mode', async () => {
+      const signingMode = TrezorKeyAgent.getSigningMode(simpleTx);
+      expect(signingMode).toEqual(Trezor.PROTO.CardanoTxSigningMode.ORDINARY_TRANSACTION);
+    });
+
+    it('can detect pool registrations signing mode', async () => {
+      const signingMode = TrezorKeyAgent.getSigningMode({
+        ...simpleTx,
+        certificates: [poolRegistrationCertificate]
+      });
+      expect(signingMode).toEqual(Trezor.PROTO.CardanoTxSigningMode.POOL_REGISTRATION_AS_OWNER);
+    });
+
+    it('can detect plutus transaction signing mode', async () => {
+      const signingMode = TrezorKeyAgent.getSigningMode(validPlutusTx);
+      expect(signingMode).toEqual(Trezor.PROTO.CardanoTxSigningMode.PLUTUS_TRANSACTION);
+    });
+
+    it('can detect multisig transaction signing mode', async () => {
+      const signingMode = TrezorKeyAgent.getSigningMode(validMultisigTx);
+      expect(signingMode).toEqual(Trezor.PROTO.CardanoTxSigningMode.MULTISIG_TRANSACTION);
+    });
+
+    describe('not valid signing mode', () => {
+      it('ordinary transaction signing mode is not valid if it contains a pool registration certificate', async () => {
+        const signingMode = TrezorKeyAgent.getSigningMode({
+          ...simpleTx,
+          certificates: [poolRegistrationCertificate]
+        });
+        expect(signingMode).not.toEqual(Trezor.PROTO.CardanoTxSigningMode.ORDINARY_TRANSACTION);
```

Comment:
What should it be equal to? I suggest improving assertions for all `not valid signing mode` tests

Reply:
Discussed: 
```
// Fn will always return the first matching type depending on the provided data
// Data is furtherly checked on Trezor side
```

Reply:
We learned that PLUTUS_TRANSACTION is the broadest signing mode, so we should update assertions to assert that it equals to it

Reply:
@mkazlauskas Unnecessary test removed in [ba1bb37](https://github.com/input-output-hk/cardano-js-sdk/pull/1017/commits/ba1bb379bcf8422e22ed9c49fc37c663b2a3a604), like we agreed on

Reply:
Thanks! Are you planning to update assertions of tests that are still relevant and should actually be plutus mode?

Reply:
yes, let's do that. I will walk through negative assertions one more time and instead of `.not.toEqual` assertion use PLUTUS signing mode where that makes sense

Reply:
Assertions updated in [bdb6efb](https://github.com/input-output-hk/cardano-js-sdk/pull/1017/commits/bdb6efb9f8d8f08b76788f7f6a845cca709b7b65).
p.s. I removed the unnecessary, negative, assertions since they are already checked in e.g `can detect pool registrations signing mode`